app/main.py

```diff
@@ -1,3 +1,47 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other):
+        return Distance(km=self.km + self.check_argument(other))
+
+    def __iadd__(self, other):
+        self.km += self.check_argument(other)
+        return self
+
+    def __mul__(self, other):
+        return Distance(self.km * self.check_argument(other))
```

Comment:
`other` can not be of type `Distance` in this method, because if you multiply km * km you get square km